### PR TITLE
Added limit exceed error message with new limit

### PR DESCRIPTION
**Pull request by dhairyasenjaliya**
### Fixed Issues

$ https://github.com/Expensify/App/issues/18647
PROPOSAL: https://github.com/Expensify/App/issues/18647#issuecomment-1540620661

### Tests

Test case send money description 

1. Login in to app 
2. Press FAB -> Select send money -> add money 
3. Select any contact 
4. Choose Description 
5. Enter more then 500 characters 
6. Notice input should not allow to exceed 500 characters  limit 

Test case edit description 

1. Login in to app 
2. Select any conversation that has money sent 
3. Go to description try to edit 
4. Notice input should not allow to exceed 500 characters  limit 

Test case Request money description 

1. Login in to app 
2. Press FAB -> Select request money -> select manual 
3. Select any contact 
4. Choose Description 
5. Enter more then 500 characters 
6. Notice input should not allow to exceed 500 characters  limit 

Test case workspace name 

1. Login in to app 
2. Press FAB -> new room 
3. Enter more then 100 characters on workspace name field 
4. Notice input should not allow to exceed 100 characters  limit 

Test case for workspace name 

1. Login in to app 
2. Press on avatar icon  -> select workspace 
3. Select any. Workspace or create new -> press on workspace name to open setting page 
4. Enter more then 100 characters on workspace name field 
5. Notice input should not allow to exceed 100 characters  limit 

Test case for task name

1. Login in to app 
2. Press on FAB  -> assign task
3. Enter more then 100 characters on task name field 
4. Notice input should not allow to exceed 100 characters  limit 

Test case for task description

1. Login in to app 
2. Press on FAB  -> assign task
3. Enter more then 100 characters on task name field 
4. Notice input should not allow to exceed 100 characters  limit 

~Test case for welcome message for room~

~1. Login in to app~
~2. Navigate to any room~
~3. Open room setting -> select Welcome message~
~4. Notice input should not allow to exceed 500 characters  limit~ 

Test case for legal name

1. Login in to app 
2. Navigate to settings -> go to profile -> go to personal details page -> go to legal name page 
3. check first and last name can only add upto 100 characters 

- [x] Verify that no errors appear in the JS console

### QA Steps

Same as above
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

workspace
Screenshot 2023-12-29 at 12 14 33 AM
Screenshot 2023-12-29 at 12 14 40 AM
Screenshot 2023-12-29 at 12 16 18 AM
sendmoney

</details>

<details>
<summary>MacOS: Desktop</summary>

workspace
Screenshot 2023-12-29 at 12 14 33 AM
Screenshot 2023-12-29 at 12 14 40 AM
Screenshot 2023-12-29 at 12 16 18 AM
sendmoney

</details>

### CODE REVIEW

**Comment by mananjadhav** [comment]
@dhairyasenjaliya Can you please update the Test Steps/QA Steps to cover all features where we applied the fix?

**Comment by dhairyasenjaliya** [comment]
@mananjadhav updated steps & screenshots

File: src/CONST.ts [hunk 1]
```diff
@@ -1488,7 +1488,10 @@ const CONST = {
     FORM_CHARACTER_LIMIT: 50,
     LEGAL_NAMES_CHARACTER_LIMIT: 150,
     LOGIN_CHARACTER_LIMIT: 254,
-    WORKSPACE_NAME_CHARACTER_LIMIT: 80,
+
+    TITLE_CHARACTER_LIMIT: 100,
+    SUPPORTING_CHARACTER_LIMIT: 500,
```
Comment (mananjadhav) [inline src/CONST.ts:1493]: I am wondering if this should be `SUPPORTING_TEXT` or `DESCRIPTION`.

@youssef-lr  wdyt?

**Comment by dhairyasenjaliya** [comment]
@mananjadhav  @youssef-lr do we need to handle the legal name as well?

**Comment by mananjadhav** [comment]
> @mananjadhav @youssef-lr do we need to handle the legal name as well?

Yes. We closed one the linked issue stating we'll take care of it here.

**Comment by dhairyasenjaliya** [comment]
Okay and any other remaining @mananjadhav

**Comment by mananjadhav** [comment]
> Okay and any other remaining @mananjadhav

I am checking through the issues.

**Comment by mananjadhav** [comment]
@dhairyasenjaliya did we cover legal name? I can't find new commits.

@youssef-lr @dhairyasenjaliya  any comments on this [one](https://github.com/Expensify/App/pull/33601/files#r1441747238)?

**Comment by dhairyasenjaliya** [comment]
@mananjadhav legal names added and steps also updated

**Comment by mananjadhav** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by mananjadhav** [comment]
@dhairyasenjaliya @youssef-lr I can see one potential issue:

When more than one fields have the validation error, we only show the error to the first field. Is that the expected behavior? Because we don't have the validation in the current form we can't make out. 

Screenshot 2024-01-16 at 10 59 53 PM


I have other forms where both the fields show the error.
image

**Comment by dhairyasenjaliya** [comment]
@mananjadhav Yeah we should stick with the current approach to show both error validations I have added that on the new commit

**Comment by mananjadhav** [comment]
Is that the only place? Do we need to worry about any other places?

**Comment by dhairyasenjaliya** [comment]
I checked its only for one place which I just updated @mananjadhav

**Comment by mananjadhav** [comment]
@dhairyasenjaliya One more issue, the latin error message doesn't get cleared in the legal name.


https://github.com/Expensify/App/assets/3069065/1b5b1dde-010c-4adf-97f3-68f51827cff4

**Comment by dhairyasenjaliya** [comment]
@mananjadhav fixed on latest commit

**Comment by mananjadhav** [comment]
Thanks @dhairyasenjaliya the above listed bugs are resolved in the latest push.

**Review by mananjadhav — APPROVED** [review]

**Comment by mananjadhav** [comment]
@dhairyasenjaliya We've got Typescript check failing. Can you merge the latest main to see if it fixes it?

**Comment by mananjadhav** [comment]
@youssef-lr quick bump.

**Comment by youssef-lr** [comment]
Reviewing today

**Comment by youssef-lr** [comment]
Sorry folks couldn't get to it last week, I'm on it today or tomorrow.

File: src/CONST.ts (hunk 1 above)
Comment (youssef-lr) [reply src/CONST.ts:1524]: I think `DESCRIPTION_LIMIT` is more consistent with what we have in the backend.

**Comment by youssef-lr** [comment]
@dhairyasenjaliya left a small comment, let's get those conflicts fixed and merge this!

File: src/CONST.ts (hunk 1 above)
Comment (dhairyasenjaliya) [reply src/CONST.ts]: done

**Comment by youssef-lr** [comment]
Lots of checks failing.

**Comment by dhairyasenjaliya** [comment]
@youssef-lr This is globally failing context - https://expensify.slack.com/archives/C01GTK53T8Q/p1706808000490579

File: src/pages/ReportWelcomeMessagePage.tsx [hunk 2]
```diff
@@ -50,6 +50,21 @@ function ReportWelcomeMessagePage({report, policy}: ReportWelcomeMessagePageProp
         setWelcomeMessage(value);
     }, []);
 
+    /**
+     * @param {Object} values
+     * @param {String} values.title
+     * @returns {Object} - An object containing the errors for each inputID
+     */
+    const validate = useCallback((values) => {
```
Comment (youssef-lr) [inline src/pages/ReportWelcomeMessagePage.tsx:58]: I'm seeing three TS errors in this block 

Screenshot 2024-02-01 at 20 02 20
Screenshot 2024-02-01 at 20 02 16
Screenshot 2024-02-01 at 20 02 12

**Comment by dhairyasenjaliya** [comment]
hey @youssef-lr all conflicts fixed please check

**Comment by youssef-lr** [comment]
I remember we had changes in `ReportWelcomeMessagePage`, we also changed this to be max 500, right now it's `10000`

**Comment by dhairyasenjaliya** [comment]
@youssef-lr on latest main it was removed for welcome message

**Comment by youssef-lr** [comment]
@dhairyasenjaliya there are conflicts, once fixed I can approve & merge.

**Comment by dhairyasenjaliya** [comment]
@youssef-lr all done please merge once you get a chance

**Review by youssef-lr — APPROVED** [review]

**Comment by lanitochka17** [comment]
This PR is failing because of issue https://github.com/Expensify/App/issues/36342

The issue is reproducible in: All environments


https://github.com/Expensify/App/assets/78819774/8ff0637d-a0fc-47e8-864e-2ac1213aa1a5

**Comment by youssef-lr** [comment]
@dhairyasenjaliya can you look into this asap please?

**Pull request merged**